### PR TITLE
[New Rule] AWS Default Policy Version Modified

### LINKED ISSUE #1500: [New Rule] AWS Default Policy Version Modified
## Description

Identifies when a default policy version is undone by reverting the previous previous version. An adversary could set an old version of the policy as the default policy and then gain full admin privilege over the AWS account.

### Target indexes

`filebeat-*, logs-aws*`

### Platforms

AWS Cloudtrail

### Query

```
event.dataset:aws.cloudtrail and event.provider:iam.amazonaws.com and 
event.action:SetDefaultPolicyVersion and event.outcome:success
```

### MITRE

| Tactic | Technique ID | Technique Name | Sub-Technique Name |
| ------ | ------------ | -------------- | ------------------ |
|  |              |                |                    |
|        |              |                |                    |

### **Adversary Possible Action**

- https://github.com/RhinoSecurityLabs/cloudgoat/tree/master/scenarios/iam_privesc_by_rollback

`aws configure --profile <Placeholder>`

`aws iam list-attached-user-policies --user-name <Placeholder> --profile <Placeholder>`

`aws iam list-policy-versions --policy-arn <generatedARN>/cg-<Placeholder>-policy --profile <Placeholder>`

`aws iam get-policy-version --policy-arn <generatedARN>/cg-<Placeholder>-policy --version-id <versionID> --profile <Placeholder>`

`aws iam set-default-policy-version --policy-arn <generatedARN>/cg-<Placeholder>-policy --version-id <versionID> --profile <Placeholder>`

### PR DESCRIPTION
**Pull request by austinsonger**
## Issues

Resolves #1500

## Contributor checklist

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)?

### CODE REVIEW

File: rules/integrations/aws/privilege_escalation_default_policy_version_modified.toml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+creation_date = "2021/10/03"
+maturity = "production"
+updated_date = "2021/10/03"
+integration = "aws"
+
+[rule]
+author = ["Austin Songer"]
+description = """
+Identifies when a default policy version is undone by reverting the previous previous version. An adversary could 
+set an old version of the policy as the default policy and then gain full admin privilege over the AWS account.
```
Comment (w0rk3r) [inline rules/integrations/aws/privilege_escalation_default_policy_version_modified.toml:11]:
```suggestion
Identifies when an IAM policy default version is modified. An adversary could change the IAM policy's default version
and potentially gain full administrator access to the AWS account.
```

File: rules/integrations/aws/privilege_escalation_default_policy_version_modified.toml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+"""
+false_positives = [
+    """
+    Default Policy Version could be modified by a system administrator. Verify whether the user identity,
```
Comment (w0rk3r) [inline rules/integrations/aws/privilege_escalation_default_policy_version_modified.toml:15]:
```suggestion
    The IAM Policy Default Version could be modified by a system administrator. Verify whether the user identity, 
```

File: rules/integrations/aws/privilege_escalation_default_policy_version_modified.toml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+rule_id = "5d8aafc2-8901-49ce-a786-4aa64f49efcf"
+severity = "low"
+tags = ["Elastic", "Cloud", "AWS", "Continuous Monitoring", "SecOps", "Identity and Access"]
+timestamp_override = "event.ingested"
+type = "query"
+
+
+query = '''
+event.dataset:aws.cloudtrail and event.provider:iam.amazonaws.com and 
+event.action:SetDefaultPolicyVersion and event.outcome:success
```
Comment (w0rk3r) [inline rules/integrations/aws/privilege_escalation_default_policy_version_modified.toml:42]: @austinsonger we are going to need of an event of this action to proceed.

**Review by w0rk3r — CHANGES_REQUESTED** [review]

**Comment by w0rk3r** [comment]
Closing this one due to the lack of requested events for testing and confirmation of effectiveness. Please feel free to open a new PR with the requested/related events if we progress on this. I am leaving the Issue open for now.

**Comment by austinsonger** [comment]
@w0rk3r Can you reopen this, so I can close it because I can reopen it. Because if you close it, I can't reopen the Pull Request.

**Pull request closed without merging**